Lib/glyphsLib/builder.py

```diff
@@ -695,17 +695,44 @@ def add_groups_to_ufo(ufo, kerning_groups):
 
 def build_gdef(ufo):
     """Build a table GDEF statement for ligature carets."""
-    carets = {}
+    bases, ligatures, marks, components, carets = set(), set(), set(), set(), {}
     for glyph in ufo:
+        has_caret = False
         for anchor in glyph.anchors:
             name = anchor.get('name')
             if name and name.startswith('caret_') and 'x' in anchor:
+                has_caret = True
                 carets.setdefault(glyph.name, []).append(str(anchor['x']))
-    if not carets:
+        glyphinfo = glyphsLib.glyphdata.get_glyph(glyph.name)
+        if glyphinfo:
+            category, subCategory = glyphinfo.category, glyphinfo.subCategory
+        else:
+            category, subCategory = None, None
+        # TODO: Find out how Glyphs.app assigns ligature components.
+        # Currently, we leave components empty but that is probably wrong.
+        # https://github.com/googlei18n/glyphsLib/issues/99
+        if has_caret or subCategory == 'Ligature':
+            ligatures.add(glyph.name)
+        elif category == 'Mark' and (subCategory == 'Nonspacing' or
+                                     subCategory == 'Spacing Combining'):
+            # https://github.com/googlei18n/ufo2ft/issues/91#issuecomment-266530863
+            marks.add(glyph.name)
+        else:
+            bases.add(glyph.name)
```

Comment:
I guess that means any glyph which is neither Ligature nor Mark is currently implied to be Base?